### DISCUSSION TITLE
Is it possible to connect to an existing contract (without deploying)?

### DISCUSSION THREAD

**Discussion by cryptotester** [scaffold-eth/scaffold-eth-2#895] [Q&A] [answered]
Hi, I'm familiar with hardhat & smart contracts, less familiar with frontend, apologies for the noob question.

I read I can configure the chain in the `scaffold.config.ts` but I guess the frontend expects the contract to be deployed using the deploy script?

I have a contract I deployed already on a testnet, I copied already the contract in `packages/hardhat/contracts`.

My goal was to generate a basic frontend using scaffold-eth-2.

If I understood well how it works, it generates (scaffolds) automatically all calls to the contract to read all existing public variables, and creates the write methods, just like etherscan does for verified contracts? This would be perfect for me...

I just need to know where to specify the address of my existing contract on testnet, and connect to it.

Should be easy, just didn't find a "recipe" for it, as I guess the normal workflow would be to deploy using the internal system and not do as I'm doing, to copy over a contract from another repo, and connect to an existing deployment on a testnet (or mainnet) but it would be very handy.

Thank you.

**Comment by carletex** [comment, marked as answer]
Hey @cryptotester 

I think you want to use `externalContracts`. Read more about it here: https://docs.scaffoldeth.io/hooks/

let me know if it works!

**Reply by cryptotester** [reply]
Thanks a lot man, wow that was so easy, didn't even need to copy over the contract to the repo, which makes sense because the ABI is all what's needed to interact with a contract. This is just awesome!

I may actually make a video demonstrating how easy and fgast it is to get a frontend thanks to this amazing project. Cheers

**Reply by carletex** [reply]
Please share the video if you do!! ;)
